x-pack/plugins/alerts/server/invalidate_pending_api_keys/task.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Logger, ISavedObjectsRepository } from 'kibana/server';
+import { InvalidateAPIKeyParams, SecurityPluginSetup } from '../../../security/server';
+import {
+  RunContext,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '../../../task_manager/server';
+import { InvalidateAPIKeyResult } from '../alerts_client';
+import { AlertsConfig } from '../config';
+import { InvalidatePendingApiKey } from '../types';
+
+const TASK_TYPE = 'alerts_invalidate_api_keys';
+export const TASK_ID = `Alerts-${TASK_TYPE}`;
+
+const invalidateAPIKey = async (
+  params: InvalidateAPIKeyParams,
+  securityPluginSetup?: SecurityPluginSetup
+): Promise<InvalidateAPIKeyResult> => {
+  if (!securityPluginSetup) {
+    return { apiKeysEnabled: false };
+  }
+  const invalidateAPIKeyResult = await securityPluginSetup.authc.invalidateAPIKeyAsInternalUser(
+    params
+  );
+  // Null when Elasticsearch security is disabled
+  if (!invalidateAPIKeyResult) {
+    return { apiKeysEnabled: false };
+  }
+  return {
+    apiKeysEnabled: true,
+    result: invalidateAPIKeyResult,
+  };
+};
+
+export function initializeAlertsInvalidateApiKeys(
```

Comment:
**optional nit:** the naming on some of these functions wasn't clear to me at first, I wonder if something like the following would work?
- `initializeAlertsInvalidateApiKeys` -> `initializeApiKeyInvalidator`
- `scheduleAlertsInvalidateApiKeys` -> `scheduleApiKeyInvalidatorTask` (function could merge with `scheduleTasks`)
- `registerAlertsInvalidateApiKeysTask` -> `registerApiKeyInvalitorTaskDefinition`